libs/ledger-live-common/src/families/ethereum/modules/erc20.ts

```diff
@@ -144,3 +158,40 @@ const erc20approve: ModeModule = {
 export const modes: Record<Modes, ModeModule> = {
   "erc20.approve": erc20approve,
 };
+
+export const fetchERC20Tokens = async () => {
+  let tokens: TokenCurrency[];
+
+  try {
+    const { data } = await network({
+      url: `${getEnv("DYNAMIC_CAL_BASE_URL")}/erc20.json`,
+    });
+
+    tokens = data.map(convertERC20);
+  } catch (e: any) {
+    log("preload-erc20", `failed to preload erc20 ${e.toString()}`);
+    tokens = [];
+  }
+
+  return tokens;
+};
+
+export async function preload(
+  currency: CryptoCurrency
+): Promise<TokenCurrency[] | null | undefined> {
+  if (currency.id !== "ethereum") {
+    return Promise.resolve(null);
+  }
+
+  const tokens = await fetchERC20Tokens();
+  addTokens(tokens);
```

Comment:
correct me if i'm wrong but i think addTokens don't "dedup" and multiple times the same token could be added as preload can run more than once?